File: src/data/topicZones.ts
/**
 * topicZones.ts
 *
 * Visual theme for each of the 10 TOPICS (not "districts"). Each topic becomes
 * one isometric town zone in the vertical journey, with its own pastel ground
 * tint, accent colour and a small integrated topic banner.
 *
 * Topic names come straight from the CSV `topic` column.
 */
export interface TopicZone {
  index: number; // 1-based, matches Subtopic.topicIndex
  name: string;
  emoji: string;
  /** Ground tint (the grass / land), kept soft and varied — never plain green. */
  groundTop: string;
  groundBottom: string;
  accent: string;
}

export const TOPIC_ZONES: TopicZone[] = [
  { index: 1, name: 'Everyday Life', emoji: '🏛️', groundTop: '#E8F3DC', groundBottom: '#D6E9C4', accent: '#5BA64E' },
  { index: 2, name: 'My Family', emoji: '🏠', groundTop: '#FBEFE0', groundBottom: '#F6E0C6', accent: '#E2904F' },
  { index: 3, name: 'Talk at Restaurant', emoji: '🍽️', groundTop: '#FaEbe2', groundBottom: '#F4D8C7', accent: '#D9714A' },
  { index: 4, name: 'Talk to Friends', emoji: '🎬', groundTop: '#F1ECFA', groundBottom: '#E2D6F3', accent: '#9168C9' },
  { index: 5, name: 'Travel', emoji: '🧭', groundTop: '#E4F1F2', groundBottom: '#CDE6E6', accent: '#3FA39B' },
  { index: 6, name: 'At Work', emoji: '🏢', groundTop: '#E9EDF6', groundBottom: '#D7DEEE', accent: '#5C6BC0' },
  { index: 7, name: 'Shopping', emoji: '🛍️', groundTop: '#FBF1DE', groundBottom: '#F6E3BE', accent: '#E0A526' },
  { index: 8, name: 'Social Interactions', emoji: '🎉', groundTop: '#FCEDF3', groundBottom: '#F7D7E5', accent: '#E0699A' },
  { index: 9, name: 'At the Hotel', emoji: '🏨', groundTop: '#ECE9F6', groundBottom: '#DBD4EF', accent: '#7E6BD0' },
  { index: 10, name: 'At the Airport', emoji: '✈️', groundTop: '#E6F2FA', groundBottom: '#CCE6F6', accent: '#3FA0D8' },
];

export function topicZoneOf(topicIndex: number): TopicZone {
  return TOPIC_ZONES[topicIndex - 1];
}

File: speakx_feature/english_town_app/lib/features/english_town/domain/town/topic_zones.dart
import 'package:flutter/material.dart';

/// Visual theme per TOPIC zone in the vertical journey (ported from
/// src/data/topicZones.ts). Only the 3 topics present in the first 20 lessons
/// are exercised, but all are included.
class TopicZone {
  final int index;
  final String name;
  final String emoji;
  final Color groundTop;
  final Color groundBottom;
  final Color accent;

  const TopicZone({
    required this.index,
    required this.name,
    required this.emoji,
    required this.groundTop,
    required this.groundBottom,
    required this.accent,
  });
}

const List<TopicZone> kTopicZones = [
  TopicZone(index: 1, name: 'Everyday Life', emoji: '🏛️', groundTop: Color(0xFFE8F3DC), groundBottom: Color(0xFFD6E9C4), accent: Color(0xFF5BA64E)),
  TopicZone(index: 2, name: 'My Family', emoji: '🏠', groundTop: Color(0xFFFBEFE0), groundBottom: Color(0xFFF6E0C6), accent: Color(0xFFE2904F)),
  TopicZone(index: 3, name: 'Talk at Restaurant', emoji: '🍽️', groundTop: Color(0xFFFAEBE2), groundBottom: Color(0xFFF4D8C7), accent: Color(0xFFD9714A)),
  TopicZone(index: 4, name: 'Talk to Friends', emoji: '🎬', groundTop: Color(0xFFF1ECFA), groundBottom: Color(0xFFE2D6F3), accent: Color(0xFF9168C9)),
  TopicZone(index: 5, name: 'Travel', emoji: '🧭', groundTop: Color(0xFFE4F1F2), groundBottom: Color(0xFFCDE6E6), accent: Color(0xFF3FA39B)),
];

TopicZone topicZoneOf(int topicIndex) => kTopicZones[topicIndex - 1];
